zetaclient/chains/bitcoin/observer/event.go

```diff
@@ -0,0 +1,253 @@
+package observer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	cosmosmath "cosmossdk.io/math"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/pkg/coin"
+	"github.com/zeta-chain/node/pkg/constant"
+	"github.com/zeta-chain/node/pkg/crypto"
+	"github.com/zeta-chain/node/pkg/memo"
+	crosschaintypes "github.com/zeta-chain/node/x/crosschain/types"
+	"github.com/zeta-chain/node/zetaclient/compliance"
+	"github.com/zeta-chain/node/zetaclient/config"
+	"github.com/zeta-chain/node/zetaclient/logs"
+)
+
+// InboundProcessability is an enum representing the processability of an inbound
+type InboundProcessability int
+
+const (
+	// InboundProcessabilityGood represents a processable inbound
+	InboundProcessabilityGood InboundProcessability = iota
+
+	// InboundProcessabilityDonation represents a donation inbound
+	InboundProcessabilityDonation
+
+	// InboundProcessabilityComplianceViolation represents a compliance violation
+	InboundProcessabilityComplianceViolation
+)
+
+// BTCInboundEvent represents an incoming transaction event
+type BTCInboundEvent struct {
+	// FromAddress is the first input address
+	FromAddress string
+
+	// ToAddress is the ZEVM receiver address
+	ToAddress string
+
+	// Value is the amount of BTC
+	Value float64
+
+	// DepositorFee is the deposit fee
+	DepositorFee float64
+
+	// MemoBytes is the memo of inbound
+	MemoBytes []byte
+
+	// MemoStd is the standard inbound memo if it can be decoded
+	MemoStd *memo.InboundMemo
+
+	// BlockNumber is the block number of the inbound
+	BlockNumber uint64
+
+	// TxHash is the hash of the inbound
+	TxHash string
+}
+
+// Processability returns the processability of the inbound event
+func (event BTCInboundEvent) Processability() InboundProcessability {
```

Comment:
nit: even when we're not modifying anything in `event` I'd suggest to use it as a pointer receiver, to maintain consistency.

Reply:
It used to be a point receiver (I just want to avoid copy). Seems we three of us (Lol!) had different ideas on this line. I have no preference except avoid copying.

Reply:
You and me vote on pointer, so here I reverted to use pointer: https://github.com/zeta-chain/node/pull/3025/commits/d0e7b08878d1c5a502a3ab487795538161205188